onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -83,40 +108,48 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
   int64_t N = X->Shape().NumDimensions() == 1 ? 1 : X->Shape()[0];
 
   Tensor* Y = ctx->Output(0, TensorShape({N}));
-  Tensor* Z;
-
   std::vector<int64_t> dims;
-  if (mode_ == SVM_TYPE::SVM_SVC && proba_.size() == 0)
-    dims = {static_cast<int64_t>(N), static_cast<int64_t>(class_count_ * (class_count_ - 1) / 2)};
-  else
-    dims = {static_cast<int64_t>(N), static_cast<int64_t>(class_count_)};
-  Z = ctx->Output(1, TensorShape(dims));
+  int64_t nc = (proba_.size() > 0 || vector_count_ == 0)
+                   ? class_count_
+                   : (class_count_ > 2 ? class_count_ * (class_count_ - 1) / 2 : 2);
+  dims = {static_cast<int64_t>(N), static_cast<int64_t>(nc)};
```

Comment:
>int64_t>(N) [](start = 22, length = 11)

For N and nc why int64_t need to be casted into itself. Consider casts an exception. Most of the time it is an indication of something wrong.

Reply:
fixed